INN/umbrella-currentorg
# Edit "Submit Listing" form to make user choose plan before categories

## Changes

This pull request makes the following changes:

- Dequeued the default `/business-directory-plugin/assets/js/submit-listing.js` and created a copied version of it at `currentorg/js/business-directory-plugin-submit-listing.js`.
- In the copied version of `submit-listing.js`, made some changes to how the form works to allow us to make the user select a plan first before categories.
- Adds customizations to the new copied version of `submit-listing.js` that determines how many categories are allowed to be selected based on the selected plan, and removes categories/alerts users accordingly.
- Adds customizations to the new copied verrsion of `submit-listing.js` that only allows you to select tags whose parent categories have been selected, and only select the maximum amount of tags based on the fee plan selected. Also turns this field into a select2 dropdown to mimic the category field.
- Adds a new REST API endpoint to expose all `wpbdp_categories` with their child tags, available at `/wp-json/currentorg/v1/wpbdp_categories`.

Before:
![screencapture-staging-current-flywheelsites-directory-of-services-2019-07-23-12_59_01](https://user-images.githubusercontent.com/18353636/61731521-a9882300-ad49-11e9-98c6-3e800d828a04.png)

After:
![screencapture-currentorg-dev-directory-of-services-2019-07-23-12_55_04](https://user-images.githubusercontent.com/18353636/61731451-80679280-ad49-11e9-91f1-c86868a84817.png)

## Why



For #97, #92 and #93.

## Testing/Questions

Features that this PR affects:

- Submit a Listing page/view


Questions that need to be answered before merging:

- [ ] This works/fires when the page is refreshed, the page is initially loaded, and the "Change plan/category" button is clicked. Are there any other events that trigger the form being rendered that this doesn't cover?
- [ ] Are there any instances where it allows you to add more than the expected number of categories?
- [ ] Are there any instances where it allows you to add tags whose parent categories have not been selected, or add more tags than allowed by the selected fee plan?

Steps to test this PR:

1. Go to create a listing page
2. Verify you have to choose a plan and then categories
3. Verify you can only add the correct amount of categories per plan
4. Verify only tags whose parent categories are selected are displaying
5. Verify you can only add the correct amount of tags per plan
6. Verify the rest of the form still works as expected
7. Verify submitting your listing still works as expected
